core/services/bulletprooftxmanager/orm.go

```diff
@@ -128,10 +138,75 @@ func (o *orm) FindEthTxAttempt(hash common.Hash) (*EthTxAttempt, error) {
 }
 
 // InsertEthTxAttempt inserts a new txAttempt into the database
-func (o *orm) InsertEthTxAttempt(txAttempt *EthTxAttempt) error {
-	query, args, err := o.db.BindNamed(insertIntoEthTxAttemptsQuery, txAttempt)
-	if err != nil {
-		return errors.Wrap(err, "InsertEthTxAttempt failed to BindNamed")
+func (o *orm) InsertEthTx(etx *EthTx) error {
+	if etx.CreatedAt == (time.Time{}) {
+		etx.CreatedAt = time.Now()
+	}
+	const insertEthTxSQL = `INSERT INTO eth_txes (nonce, from_address, to_address, encoded_payload, value, gas_limit, error, broadcast_at, created_at, state, meta, subject, pipeline_task_run_id, min_confirmations, evm_chain_id, access_list, simulate) VALUES (
+:nonce, :from_address, :to_address, :encoded_payload, :value, :gas_limit, :error, :broadcast_at, :created_at, :state, :meta, :subject, :pipeline_task_run_id, :min_confirmations, :evm_chain_id, :access_list, :simulate
+) RETURNING *`
+	err := postgres.NewQ(o.db).GetNamed(insertEthTxSQL, etx, etx)
+	return errors.Wrap(err, "InsertEthTx failed")
+}
+
+func (o *orm) InsertEthTxAttempt(attempt *EthTxAttempt) error {
+	const insertEthTxAttemptSQL = `INSERT INTO eth_tx_attempts (eth_tx_id, gas_price, signed_raw_tx, hash, broadcast_before_block_num, state, created_at, chain_specific_gas_limit, tx_type, gas_tip_cap, gas_fee_cap) VALUES (
+:eth_tx_id, :gas_price, :signed_raw_tx, :hash, :broadcast_before_block_num, :state, NOW(), :chain_specific_gas_limit, :tx_type, :gas_tip_cap, :gas_fee_cap
+) RETURNING *`
+	err := postgres.NewQ(o.db).GetNamed(insertEthTxAttemptSQL, attempt, attempt)
+	return errors.Wrap(err, "InsertEthTxAttempt failed")
+}
+
+func (o *orm) InsertEthReceipt(receipt *EthReceipt) error {
```

Comment:
Seems to be missing a test

Reply:
It's indirectly covered by many tests, do you think it needs one explicitly?